backend/src/main/kotlin/org/loculus/backend/controller/SubmissionController.kt

```diff
@@ -260,30 +250,34 @@ open class SubmissionController(
     fun getReleasedData(
         @PathVariable @Valid organism: Organism,
         @RequestParam compression: CompressionFormat?,
+        @RequestHeader(value = HttpHeaders.IF_NONE_MATCH, required = false) ifNoneMatch: String?,
     ): ResponseEntity<StreamingResponseBody> {
-        val headers = HttpHeaders()
-        headers.contentType = MediaType.parseMediaType(MediaType.APPLICATION_NDJSON_VALUE)
-        if (compression != null) {
-            headers.add(HttpHeaders.CONTENT_ENCODING, compression.compressionName)
+        val lastDatabaseWriteETag = releasedDataModel.getLastDatabaseWriteETag()
+        if (ifNoneMatch == lastDatabaseWriteETag) {
+            return ResponseEntity.status(HttpStatus.NOT_MODIFIED).build()
         }
 
+        val headers = HttpHeaders()
+        headers.eTag = lastDatabaseWriteETag
```

Comment:
Let's add the header to the Swagger Docs.

Reply:
It should be?

Reply:
Yes, the return header. You can simply add it to the `@ApiResponse` above.